# API Mini Project

**1. Collect data from the Franfurt Stock Exchange, for the ticker AFX_X, for the whole year 2017 (keep in mind that the date format is YYYY-MM-DD).**

In [279]:
import requests
import json
from pandas.io.json import json_normalize
import collections

In [280]:
#API Key
API_KEY = ''
params = {'api_key':API_KEY, 'start_date':'2017-01-01', 'end_date':'2017-12-31'}
URL = 'https://www.quandl.com/api/v3/datasets/FSE/AFX_X.json'

In [281]:
r = requests.get(URL, params=params)

**2. Convert the returned JSON object into a Python dictionary.**

In [282]:
r = r.json()
r = r['dataset']
type(r)

dict

In [283]:
# to access applicable data we must access the dictionary key 'data'
data = r.get('data')

**3. Calculate what the highest and lowest opening prices were for the stock in this period.**

In [284]:
open_list = []
open_column_index = 1
for row in data:
    if row[open_column_index] is not None:
        open_list.append(row[open_column_index])

max(open_list), min(open_list)

(53.11, 34.0)

Max = 53.11
Min = 34.00

**4. What was the largest change in any one day (based on High and Low price)?**

In [285]:
change_list = []
high_column_index = 2
low_column_index = 3

for row in data:
    if row[high_column_index] and row[low_column_index] is not None:
        diff = row[high_column_index] - row[low_column_index]
        change_list.append(diff)

max(change_list)

2.8100000000000023

Max = 2.81

**5. What was the largest change between any two days (based on Closing Price)?**

In [286]:
closing_list = []
closing_price_index = 4
change_list = []

for row in data:
    if row[closing_price_index] is not None:
        closing_list.append(row[closing_price_index])

length = len(closing_list)

for i in range(length):
    if i + 1 < length: 
        two_day_change = closing_list[i+1] - closing_list[i]
        two_day_change = abs(two_day_change)
        change_list.append(two_day_change)
max(change_list)

2.559999999999995

Max = 2.56

**6. What was the average daily trading volume during this year?**

In [287]:
volume_list = []
volume_column_index = 6
for row in data:
    if row[volume_column_index] is not None:
        volume_list.append(row[volume_column_index])
for i in volume_list:
    total = i+i

length = len(volume_list)
average = total/length
average

350.5882352941176

Average = 350.59